### PR TITLE
refs create 67657 via fb twitter

### PR DESCRIPTION
**Pull request by daonhat**

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
+  <div class="col-lg-5 col-lg-push-7">
+    <div class="panel panel-default">
+      <div class="panel-heading dn-panel-head">
+        <h4><%= t :login %></h4>
+      </div>
+      <div class="panel-body">
+        <%= form_for resource, as: resource_name, url: session_path(resource_name) do |f| %>
+          <div class="field dn-field">
+            <%= f.email_field :email, autofocus: true, class: "form-control",
+            placeholder: t(:your_email) %>
```
Comment (tiennh1995) [inline app/views/devise/sessions/new.html.erb]: Ident

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -18,3 +18,11 @@
 //= require bootstrap
 //= require turbolinks
 //= require_tree .
+
+$(document).on("ready", function() {
+  set_timetout();
+
```
Comment (ngtrduc) [inline app/assets/javascripts/application.js]: thua dong

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -1,27 +1,47 @@
...
+          </div>
 
-    <div class="field">
-      <%= f.label t :password %><br />
-      <%= f.password_field :password, autocomplete: "off", class: "form-control" %>
-    </div>
+          <div class="field dn-field">
+            <%= f.password_field :password, autocomplete: "off",
+            class: "form-control", placeholder: t(:your_pass) %>
```
Comment (tiennh1995) [inline app/views/devise/sessions/new.html.erb]: Ident

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -18,3 +18,11 @@
 //= require bootstrap
 //= require turbolinks
 //= require_tree .
+
+$(document).on("ready", function() {
```
Comment (ngtrduc) [inline app/assets/javascripts/application.js]: trong js dung '

**Comment by qtsoul95** [comment]
Ok

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+
+  %w(facebook twitter).each do |provider|
+    define_method provider do
+      @user = User.from_omniauth request.env["omniauth.auth"]
+      if @user.provider == "twitter"
```
Comment (ngtrduc) [inline app/controllers/users/omniauth_callbacks_controller.rb]: nguoc

**Comment by tiennh1995** [comment]
ok

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+
+  %w(facebook twitter).each do |provider|
+    define_method provider do
+      @user = User.from_omniauth request.env["omniauth.auth"]
+      if "twitter" == @user.provider 
+        nickname = env["omniauth.auth"].info.nickname.gsub(/[^0-9A-Za-z]/, "")
+        @user.email = "#{nickname}@fels.xxx"
+        @user ||= User.find_by email: @user.email
+        @user ||= User.create  email: @user.email,
```
Comment (hellovietnam93) [inline app/controllers/users/omniauth_callbacks_controller.rb]: cọnvention

File: app/models/user.rb [hunk 7]
```diff
@@ -26,4 +26,14 @@ def unfollow other_user
   def following? other_user
     following.include? other_user
   end
+
+  class << self
+    def from_omniauth auth
+      user = User.find_or_initialize_by(provider: auth.provider, uid: auth.uid)
```
Comment (hellovietnam93) [inline app/models/user.rb]: bỏ ()

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        @user.email = "#{nickname}@fels.xxx"
+        @user ||= User.find_by email: @user.email
+        @user ||= User.create email: @user.email,
+          password: @user.uid, password_confirmation: @user.uid
+        @user.save
+      end
+      @user.create_profile
+      if @user.persisted?
+        sign_in_and_redirect @user, event: :authentication
+        set_flash_message(:notice, :success, kind: provider.capitalize) if is_navigational_format?
```
Comment (hellovietnam93) [inline app/controllers/users/omniauth_callbacks_controller.rb]: BỎ ()

**Pull request merged**